### PR TITLE
Fix #268

### LINKED ISSUE #268: Rocket.Chat+ on Android: Sending message stuck in "Sending" state, displaying wrong message owner
Rocket.Chat Version: 0.55.1
Running Instances: 1
DB Replicaset OpLog: 0
Node Version: 4.8.1 (We use Official Docker image)
Android app version: 1.0.10

When sending message from Android app. Most of the time it will stuck in the "Sending" state and showing wrong owner of the message (It should be the owner of the phone, but it randomly pick someone around and show the message state as Sending).

This seems to happen when we exit the app for a while and comeback (Usually from push notification). The App works fine the first time we login to the server.

When the message was in Sending state, the message was actually already delivered (by checking on Web-site or Cordova app).

Sometime the message was stuck in "!" state and need resend, although the message was actually delivered.

The current workaround is to push "Back" button to exit the app, or kill the app and re-open it.

This didn't happen with Cordova app.

NOTE: We use Rocket.chat Gateway in push notification setting.

### PR DESCRIPTION
**Pull request by philipbrito**
@RocketChat/android

Closes #268

### CODE REVIEW

**Comment by rafaelks** [comment]
LGTM! Can you review the change @matheusjardimb, please?

File: app/src/main/java/chat/rocket/android/renderer/MessageRenderer.java [hunk 1]
```diff
@@ -76,10 +75,11 @@ public MessageRenderer timestampInto(TextView textView) {
     }
 
     switch (object.getSyncState()) {
```
Comment (leonardoaramaki) [inline app/src/main/java/chat/rocket/android/renderer/MessageRenderer.java:77]: Shouldn't it be better if we put this way:

case SyncState.FAILED:
         textView.setText(R.string.failed_to_sync);
         break;

Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/renderer/MessageRenderer.java:77]: Nice catch! But, what is the diference between FAILED and NOT_SYNCED state?
@rafaelks ?

Comment (leonardoaramaki) [reply app/src/main/java/chat/rocket/android/renderer/MessageRenderer.java:77]: @filipedelimabrito As I understand so far, NOT_SYNCED is the first state of a message before it's sent, when it's first created.

Comment (matheusjardimb) [reply app/src/main/java/chat/rocket/android/renderer/MessageRenderer.java:77]: @leonardoaramaki I agree! `NOT_SYNCED` seems to be the first state. `MessageInteractor::send` exemplifies that. The state then becomes `SYNCING` and later either `SYNCED` or `FAILED`.

**Comment by rafaelks** [comment]
@RocketChat/android is it good to merge?

**Review by leonardoaramaki — APPROVED** [review]

**Pull request merged**